### PR TITLE
spaCy v3 is not saving the best version in training loop

### PR DESCRIPTION
**Pull request by bratao**
## Description

During a training session using the latest spaCy v3 I realized that it was not saving the "best" version during training, it was saving only the latest. 
After a debug I found the error. During the training loop the check condition for is_best_checkpoint was wrong. I also realized that it was always saving the optimizer average, and also put a condition to check this. 

### Types of change
It is a bugfix to the training loop

## Checklist

- [X] I have submitted the spaCy Contributor Agreement.
- [X] I ran the tests, and all new and existing tests passed.
- [X] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by honnibal** [comment]
Thanks!

The `optimizer.averages` check shouldn't be necessary: inside `model.use_params`, it should be a no-op if the parameters are `None`. This was to avoid the annoyance of conditioning on the context manager.

**Comment by bratao** [comment]
@honnibal Got it. I remove the average check in the latest commit

File: spacy/training/loop.py [hunk 1]
```diff
@@ -87,9 +87,12 @@ def train(
             if is_best_checkpoint is not None and output_path is not None:
                 with nlp.select_pipes(disable=frozen_components):
                     update_meta(T, nlp, info)
-                with nlp.use_params(optimizer.averages):
-                    nlp = before_to_disk(nlp)
```
Comment (adrianeboyd) [inline spacy/training/loop.py:91]: The `before_to_disk` callback should be preserved in these changes.

Comment (bratao) [reply spacy/training/loop.py:91]: Done. I did not called before_to_disk in the save below because this one will always be called

**Pull request merged**